File: apple/LC205IsomorphicStrings.py
'''
利口205
'''
class Solution(object):
    def isIsomorphic(self, s, t):
        if not s or not t:
            return s==t
        if len(s) != len(t):
            return False
        dictionary = {}
        for i in range(len(s)):
            if s[i] not in dictionary:
                if t[i] not in dictionary.values():
                    dictionary[s[i]] = t[i]
                else:
                    return False
            else:
                if dictionary[s[i]] != t[i]:
                    return False
                else:
                    pass
        return True

File: apple/LC205IsomorphicStrings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s == null || t== null) return s==t;
        HashMap<Character, Character> hm = new HashMap<Character, Character>();
        for (int i=0; i < s.length(); i++) {
            char a = s.charAt(i);
            char b = t.charAt(i);
            if (hm.containsKey(a)) {
                if (!hm.get(a).equals(b)) return false;
                else continue;
            } else {
                if(!hm.containsValue(b)) hm.put(a,b);
                else return false;
            }
        }
        return true;
    }
}
